### PR TITLE
Move msbuild tasks into src folder and delete dead code

### PR DESCRIPTION
**Pull request by ViktorHofer**
Contributes to https://github.com/dotnet/runtime/issues/36519

We'll always have a need for repository local msbuild tasks. As the tools-local folder only clutters the repository root, moving them into the src folder should be a better place.

Also deleting dead task code.

### CODE REVIEW

File: src/mono/netcore/nuget/Microsoft.NET.Runtime.Android.Sample.Mono/Microsoft.NET.Runtime.Android.Sample.Mono.pkgproj [hunk 1]
```diff
@@ -3,7 +3,7 @@
   <Import Project="$([MSBuild]::GetPathOfFileAbove(Directory.Build.props))" />
 
   <ItemGroup>
-    <ProjectReference Include="$(RepoToolsLocalDir)tasks\mobile.tasks\AndroidAppBuilder\AndroidAppBuilder.csproj" />
+    <ProjectReference Include="$(RepoTasksDir)mobile.tasks\AndroidAppBuilder\AndroidAppBuilder.csproj" />
```
Comment (jkotas) [inline src/mono/netcore/nuget/Microsoft.NET.Runtime.Android.Sample.Mono/Microsoft.NET.Runtime.Android.Sample.Mono.pkgproj:6]: Do we need the `mobile.tasks` intermediate directory? Would it make more sense to have the tasks directly in the `RepoTasksDir` ?

Comment (ViktorHofer) [reply src/mono/netcore/nuget/Microsoft.NET.Runtime.Android.Sample.Mono/Microsoft.NET.Runtime.Android.Sample.Mono.pkgproj:6]: @akoeplinger @steveisok @marek-safar are more suitable to answer this :)

Comment (akoeplinger) [reply src/mono/netcore/nuget/Microsoft.NET.Runtime.Android.Sample.Mono/Microsoft.NET.Runtime.Android.Sample.Mono.pkgproj:6]: remove the intermediate directory sounds fine to me.

Comment (ViktorHofer) [reply src/mono/netcore/nuget/Microsoft.NET.Runtime.Android.Sample.Mono/Microsoft.NET.Runtime.Android.Sample.Mono.pkgproj]: done

**Comment by ViktorHofer** [comment]
Someone pls approve so that I can merge the PR tomorrow morning ;)

**Review by jkotas — APPROVED** [review]

**Review by steveisok — APPROVED** [review]

**Comment by marek-safar** [comment]
`src\tasks` sounds a bit strange to me, did you think about placing them under `src\msbuild\tasks`

**Comment by ViktorHofer** [comment]
We don't have a parent msbuild folder anywhere else, neither in eng which mostly contains msbuild files, nor in src libraries. As our primary build engine is msbuild, I don't think we need the extra msbuild folder. Or let me rephrase, what else would we put there? Other pure msbuild files like Directory.Build.props/targets/rsp actually can't be in a subfolder, otherwise they wouldn't apply.

**Comment by marek-safar** [comment]
it's just not obvious to me that `src/tasks` is about in-tree msbuild tasks support but I can live with that

**Comment by ViktorHofer** [comment]
> it's just not obvious to me that src/tasks is about in-tree msbuild tasks support but I can live with that

I will add a README.md to make that clear. Thanks for the feedback.

**Comment by ViktorHofer** [comment]
Submitted https://github.com/dotnet/runtime/pull/45754
